### ISSUE TITLE
update requiring pip script is unreliable 

### ISSUE THREAD

**Issue by dirkpetersen** [dirkpetersen/froster#54]
After I updated to the latest version, I get this once once i update again. It seems froster requires the 'pip' script. This is unreliable as per python best practices, one should use `python3 -m pip` instead of `pip` as pip my belong to another python3 install and not the one in the current path 

[peterdir@loginnode2 ~]$ froster update

Backing up older froster installation (if any)...
    Data back up at /home/users/peterdir/.local/share/froster_20240604120731.bak
    Config back up at /home/users/peterdir/.config/froster_20240604120731.bak
...older froster installation backed up

Uninstalling existing froster installation...
 using pip with /usr/bin/python3 ...
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.

 Please note that packages are installed in /home/users/peterdir/.local/lib/python...
 If you share 'HPC Toys' with others use 'htypip install ...' instead.

...froster uninstalled
Error: pipx: exit code 1

Rolling back installation...
...done

Installation failed!
Error: The update failed with exit code 1.

**Comment by victormachadoperez** [comment]
I tried several approaches for updating Froster at running time (i.e., using the "froster update" command), but I faced several issues due to venv collisions with itself.

The final update implementation, which is already available in your repo and PyPi, works as follows:

- The same installation script is in charge of updating Froster; therefore, it is updated outside running time.
- "froster update" only checks if a new Froster version is published at the PyPi repository.
  - If an update is available, it prompts the same "curl ... install.sh | bash" script command to update Froster.
- If a week has passed since the last update check, "froster update" is automatically executed at the next froster command.
  - if an update is available, then prompt the update command.
  - if don't, do not print anything.
  - In either case, store the current timestamp to support next automatic update check.
- rclone is downloaded and installed (i.e. updated) everytime froster is installed or updated.
  - This is done this way because rclone does not have a cli command to check for updates.

Also, "git" has been removed as installation dependency as the installation script installs the latests PyPi release (not the github's one)
The PyPi release is automatically published once the github release is deployed, BUT the github release has to manually be released. We can discuss why I took this approach next meeting.

**Comment by dirkpetersen** [comment]
Actually this issue is still happening, there is a 'pip uninstall' in the install.sh script. This should be replaced with 'python3 -m pip' as  people who use many different versions of python often have inconsistencies in the path and pip and python don't always belong to the same install. I was hoping pipx does not have the same problem 


```
[peterdir@fff ~]$ curl -s https://raw.githubusercontent.com/dirkpetersen/froster/main/install.sh?$(date +%s) | bash

Backing up older froster installation (if any)...
    Config back up at /home/users/peterdir/.config/froster_20240614101553.bak
...older froster installation backed up

Uninstalling existing froster installation...
 using pip with /usr/bin/python3 ...
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.

 Please note that packages are installed in /home/users/peterdir/.local/lib/python...
 If you share 'HPC Toys' with others use 'htypip install ...' instead.

...froster uninstalled

Installing latest version of froster...
  Installing from PyPi package repository
...froster installed

Installing third-party dependency: pwalk...
    Downloading pwalk files
    Compiling pwalk
    Moving pwalk to froster's binaries folder
    Cleaning up pwalk installation files
...pwalk installed

Installing third-party dependency: rclone...
    Downloading rclone files
    Extracting rclone files
    Moving rclone to froster's binaries folder
    Cleaning up rclone installation files
...rclone installed

froster v0.12.31 has been successfully installed!
```

**Comment by victormachadoperez** [comment]
The pip install command installed Froster locally for development and Github Actions testing.
When uninstalling older Froster versions, it checks if it was installed under pip or pipx environment.
The message you see is due to the pip installation check, which makes no sense for production.
I updated the local installation method to also use pipx, so no more pip command.

This change is available in the already deployed Froster v0.12.32 release
